cylc/flow/scripts/lint.py

```diff
@@ -129,10 +129,50 @@
         'url': STYLE_GUIDE + 'trailing-whitespace',
         'index': 6
     },
-    # Look for families both from inherit=FAMILY and FAMILY:trigger-all/any.
-    # Do not match inherit lines with `None` at the start.
+    # Look for families in inherit= configurations.
     re.compile(
-        r'(inherit\s*=(?!\s*None\s*(?!.*[a-z])))|(\w[a-z]\w:\w+?-a(ll|ny))'
+        r'''
+          # match all inherit statements
+          ^\s*inherit\s*=
+          # filtering out those which match only valid family names
+          (?!
+            \s*
+            # none, None and root are valid family names
+            (none|None|root|
+              (
+                # as are families named with capital letters
+                [A-Z0-9_-]+
+                # which may include Cylc parameters
+                | (<[^>]+>)
+                # or Jinja2
+                | ({[{%].*[%}]})
+                # or EmPy
+                | (@[\[{\(]).*([\]\}\)])
+              )+
+            )
+            # this can be a comma separated list
+            (
+              \s*,\s*
+              # none, None and root are valid family names
+              (none|None|root|
+                (
+                  # as are families named with capital letters
+                  [A-Z0-9_-]+
+                  # which may include Cylc parameters
+                  | (<[^>]+>)
+                  # or Jinja2
+                  | ({[{%].*[%}]})
+                  # or EmPy
+                  | (@[\[{\(]).*([\]\}\)])
+                )+
```

Comment:
Duplicate of the above logic (Regex doesn't support variables).

If re-used elsewhere we can refactor this out using string templating. Example here https://github.com/cylc/cylc-flow/blob/c5f177024b15e392fa1a0ba0f5c1d908fa6dc4f6/cylc/flow/id.py#L500

Reply:
Not really helpful.

However, this is now outdated as these two patterns don't match.

Reply:
These two are identical aren't they?

https://github.com/cylc/cylc-flow/blob/e540410d7dd1084d338206e07698d85771462ba5/cylc/flow/scripts/lint.py#L143-L154

https://github.com/cylc/cylc-flow/blob/e540410d7dd1084d338206e07698d85771462ba5/cylc/flow/scripts/lint.py#L161-L172

Reply:
It's the line above containing none, None and root which now differs.